libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/ManeuverProcessor.kt

```diff
@@ -292,27 +307,27 @@ internal object ManeuverProcessor {
             stepDistance,
             secondaryManeuver,
             subManeuver,
-            laneGuidance
+            laneGuidance,
+            PolylineUtils.decode(geometry, precision).toList().last()
```

Comment:
`PolylineUtils.decode` is returning a `List` of `Point`s and it may be expensive. Could we rely on something less expensive that doesn't need to traverse and decode the whole `geometry`? 🤔

Reply:
There is no other way at present. However, we have identified that this operation is being repeated at several places across the SDK (Routeline, Camera and now Maneuver etc.) The idea is to unblock this PR and use it to find the last point in the current step geometry. Parallely, we are going to create a new ticket whose sole purpose would be to figure out how to prepare the list of points in a step geometry and which could be referenced from anywhere in the SDK. 
**Initial thoughts: Maybe a singleton**

cc @Guardiola31337

Reply:
If it's just the last point you're after perhaps drill down into the steps. They have geometry that's much smaller.  DirectionsRoute->lastRouteLeg->lastRouteStep->geometry.

Reply:
If that doesn't work there is a private method in `MapboxRoutelineUtils` that decodes a route and caches the result. Perhaps we should think about making it available in a static way so that any part of the SDK could call it. After all, the various components are operating on the same routes.  The route line is going to initiate the caching of the geometry before anything else.

Reply:
`PolylineUtils.decode(geometry, precision).last()`
The `geometry` referred here is `step.geometry` that is being passed from the calling function. Based on my initial discussion with @LukasPaczos and my discussion with @cafesilencio today, I am gonna leave this piece of code as is and see how we can expose `MapboxRoutelineUtils::decodeRoute` for the entire SDK and not just route line.

Reply:
MapboxRoutelineUtils::decodeRoute currently decodes and caches the whole route to a LineString. If what's needed here is something different then this specific method might not be the answer but the caching mechanism it uses could be.

Reply:
Okay so I have taken a different approach. The use case is simpler in case of `Maneuver`
#### Use case
Every time a new route is fetched or the route updates, the API needs to go through the list of steps in a given leg and create a `Maneuver` object which holds the last point in every `step`
#### Pseudo code:
```
val steps = leg.steps
for (index in 0..steps.lastIndex) {
    val stepIntersections = if (index == steps.lastIndex) {
        steps[stepIndex].intersections()
    } else {
        steps[stepIndex + 1].intersections()
    }
}
val maneuverPoint: Point = stepIntersections.first().location
```
First point of a `StepIntersection` contains the starting point of the step. Hence instead of doing 
stepX.geometry.getPoints().last() I am doing stepX+1.intersections.first().point.
For the last step I am doing stepX.intersections.first().point.

cc @cafesilencio @LukasPaczos @Guardiola31337

Reply:
If this gets you the needed values then it looks less expensive than the geometry decoding.

Reply:
Yeah this for loop is already in the logic for other purposes. So I did not had to introduce it and all I do is read the step intersections.